torch/package/importer.py

```diff
@@ -184,3 +184,11 @@ def import_module(self, module_name: str) -> ModuleType:
             raise last_err
         else:
             raise ModuleNotFoundError(module_name)
+
+    def whichmodule(self, obj: Any, name: str) -> str:
+        for importer in self._importers:
+            module_name = importer.whichmodule(obj, name)
+            if module_name != "__main__":
+                return module_name
+
+        return "__main__"
```

Comment:
This is what the CPython `whichmodule` does too but I don't really understand why. 😕  Is it understood to indicate that a module wasn't found?

Reply:
honestly I am not sure, I suppose the idea is that an object without `__module__` defined that also isn't accessible from any known module *must* have been defined interactively/from a script.